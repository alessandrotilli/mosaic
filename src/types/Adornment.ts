import { IBase } from "./Base";

type IBadgeColor = "default" | "error" | "primary" | "secondary";

type IBadgeVariant = "dot" | "standard";

export interface IBadgeAdornment {
  color?: IBadgeColor;
  value: string;
  variant?: IBadgeVariant;
}

export interface IAdornment extends IBase {
  /**
   * Badge properties
   */
  badge?: IBadgeAdornment;
  /**
   * Text of tooltip
   */
  tooltip?: string;
}
