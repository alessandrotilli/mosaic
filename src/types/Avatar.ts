import { ILoadable } from "./Base";
import { Icons } from "./Icon";

export enum AvatarVariant {
  default = "circular",
  rounded = "rounded",
  squared = "square",
}

export interface IAvatar extends ILoadable {
  alt?: string;
  icon?: Icons;
  src?: string;
  text?: string;
  variant?: AvatarVariant;
}
