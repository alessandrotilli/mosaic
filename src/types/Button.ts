import { ReactElement } from "react";

import { IClickable, ILocalizable } from "./Base";
import { Icons, IIcon } from "./Icon";
import { IDisablableInput } from "./Input";

/**
 * @deprecated Use "left" | "right"
 */
export enum ButtonIconPosition {
  left = "left",
  right = "right",
}

type IButtonIconPosition = "left" | "right";

/**
 * @deprecated Use "contained" | "outlined"
 */
export enum ButtonVariants {
  contained = "contained",
  outlined = "outlined",
}

type IButtonVariants = "contained" | "outlined";

export type ButtonColor = "primary" | "success" | "secondary" | "info" | "error" | "warning";

export type IButtonIcon = Pick<IIcon, "rotate"> & {
  component?: ReactElement;
  name?: Icons;
  position?: IButtonIconPosition | ButtonIconPosition;
};

export interface IMUIButtonIcon {
  endIcon?: ReactElement;
  startIcon?: ReactElement;
}

export interface IBaseButton extends IClickable {
  /**
   * Text displayed
   */
  label: string;
}

export interface IButton extends IBaseButton, IDisablableInput, ILocalizable {
  /**
   * Component elevated state, adds shadowing
   */
  elevated?: boolean;
  /**
   * Icon displayed
   */
  icon?: IButtonIcon;
  /**
   * Variant for button shape
   */
  variant?: IButtonVariants | ButtonVariants;
  /**
   * Button color
   */
  color?: ButtonColor;
}
