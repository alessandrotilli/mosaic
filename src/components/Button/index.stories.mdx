import { action } from '@storybook/addon-actions';
import { Meta,Preview, Props, Story} from '@storybook/addon-docs/blocks';
import { boolean, object, select, text } from '@storybook/addon-knobs';
import { ButtonIconPosition, ButtonVariants } from '../../types/Button';
import { Icons } from '../../types/Icon';
import Button from '.';

<Meta title="Button-MDX" component={Button} />

# Button

<Preview>
  <Story name="Default">
    <Button
      dataCy={text('data-cy', 'button-identifier')}
      elevated={boolean('elevated', false)}
      icon={object('icon', { name: Icons.send, position: ButtonIconPosition.left})}
      label={text('label', 'Example')}
      onClick={action('Click on Button')}
      variant={select('variant', ButtonVariants, ButtonVariants.contained)}
    />
  </Story>
</Preview>

# Props

<Props of={Button} />