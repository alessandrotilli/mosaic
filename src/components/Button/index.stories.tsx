import React from "react";
import MUIStyleIcon from "@mui/icons-material/Style";
import { expect, jest } from "@storybook/jest";
import { Meta, StoryObj } from "@storybook/react";
import { configure, userEvent, within } from "@storybook/testing-library";

import { Icons } from "../../types/Icon";
import { getAllComposedDataCy } from "../../utils";
import { logInfo } from "../../utils/logger";
import { localeDecorator, MessageMock } from "../../utils/mocks/LocaleMock";
import getDocsPage from "../../utils/stories";

import { DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS, LocalizedButton, SUBPARTS_MAP } from ".";

configure({ testIdAttribute: "data-cy" });

const COMPONENT_NAME = "Button";
LocalizedButton.displayName = COMPONENT_NAME;

const meta = {
  title: "Inputs/Button",
  component: LocalizedButton,
  decorators: [localeDecorator],
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Button Component",
          url: "https://mui.com/components/buttons/",
        },
        component: COMPONENT_NAME,
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          dataCyShortcut: DATA_CY_SHORTCUT,
          subpartsSuffixes: getAllComposedDataCy(SUBPARTS_MAP),
        },
        localizableProps: LOCALIZABLE_PROPS,
      }),
    },
  },
} satisfies Meta<typeof LocalizedButton>;

export default meta;
type Story = StoryObj<typeof meta>;

const onClickMock = jest.fn(() => logInfo(COMPONENT_NAME, "onClick handler"));

export const Primary: Story = {
  args: {
    label: "Button",
    onClick: onClickMock,
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const button = canvas.getByTestId(DATA_CY_DEFAULT);
    if (button) {
      await userEvent.click(button);
      await expect(onClickMock).toHaveBeenCalledTimes(onClickMock.mock.calls.length);
    }
  },
};

export const Color = {
  args: {
    ...Primary.args,
    color: "success",
  },
};

export const Disabled: Story = {
  args: {
    ...Primary.args,
    disabled: true,
  },
};

export const Elevated: Story = {
  args: {
    ...Primary.args,
    elevated: true,
  },
};

export const Icon: Story = {
  args: {
    ...Primary.args,
    icon: {
      name: Icons.send,
    },
  },
};

export const IconCustom: Story = {
  args: {
    ...Primary.args,
    icon: {
      component: <MUIStyleIcon />,
    },
  },
};

export const IconRight: Story = {
  args: {
    ...Primary.args,
    icon: {
      name: Icons.send,
      position: "right",
    },
  },
};

export const IconRotate: Story = {
  args: {
    ...Disabled.args,
    icon: {
      name: Icons.refresh,
      rotate: true,
    },
    label: "Loading",
  },
};

export const Localized: Story = {
  args: {
    ...Primary.args,
    localized: true,
    label: MessageMock.button,
  },
};

export const Styled: Story = {
  args: {
    ...Primary.args,
    style: {
      backgroundColor: "red",
      color: "white",
    },
  },
};

export const VariantOutlined: Story = {
  args: {
    ...Primary.args,
    variant: "outlined",
  },
};
