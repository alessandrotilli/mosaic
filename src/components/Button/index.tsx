import React, { FC, ReactElement } from "react";
import MUIButton from "@material-ui/core/Button";
import Icon from "../Icon";
import { ButtonIconPosition, IButtonIcon, IButton, ButtonVariants } from "../../types/Button";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";

interface IMUIButtonIcon {
  endIcon?: ReactElement;
  startIcon?: ReactElement;
}

const getIcons = (dataCy: string, iconConfig?: IButtonIcon): IMUIButtonIcon => {
  const muiIcon = {};
  if (!iconConfig) {
    return muiIcon;
  }

  const { name, position } = iconConfig;
  const icon = <Icon dataCy={`${dataCy}-icon`} name={name} />;
  return position === ButtonIconPosition.right
    ? {
        endIcon: icon,
      }
    : { startIcon: icon };
};

export const DATA_CY_DEFAULT = "button";
export const DATA_CY_SHORTCUT = "label";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [{ name: "label", type: "string" }];

// TODO: handle color
const Button: FC<IButton> = ({
  dataCy = DATA_CY_DEFAULT,
  disabled = false,
  elevated = false,
  icon = undefined,
  label,
  onClick,
  variant = ButtonVariants.contained,
}) => {
  return (
    <MUIButton
      color="primary"
      data-cy={dataCy}
      disabled={disabled}
      disableElevation={!elevated}
      onClick={onClick}
      variant={variant}
      {...getIcons(dataCy, icon)}
    >
      {label}
    </MUIButton>
  );
};

export const ButtonWithProps = Button;

export default localized(Button, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});
