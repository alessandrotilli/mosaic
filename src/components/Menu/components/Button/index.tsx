import React, { useMemo } from "react";

import { Icons } from "../../../../types/Icon";
import { IMenuButton } from "../../../../types/Menu";
import { logWarn } from "../../../../utils/logger";
import Button from "../../../Button";
import IconButton from "../../../IconButton";

interface MenuButtonProps extends IMenuButton {
  onMenu: () => void;
}

const MenuButton = ({ dataCy, icon: externalIcon, label, onMenu, style, type = "button" }: MenuButtonProps) => {
  const icon = useMemo(() => {
    if (!externalIcon) {
      logWarn("Menu", "Icon is not set, fallback to Icons.menu");
      return Icons.menu;
    }

    return externalIcon;
  }, [externalIcon]);

  const buttonIcon = useMemo(() => (typeof icon === "string" ? { name: icon } : { component: icon }), [icon]);

  if (type === "icon") {
    return <IconButton dataCy={dataCy} icon={icon} onClick={onMenu} style={style} tooltip={label} />;
  }

  return <Button dataCy={dataCy} icon={buttonIcon} label={label} onClick={onMenu} style={style} />;
};

export default MenuButton;
