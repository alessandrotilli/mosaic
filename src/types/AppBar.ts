import { ILocalizable } from "./Base";
import { IBaseButton } from "./Button";
import { IBaseIconButton } from "./IconButton";

export interface IAppBar extends ILocalizable {
  actions?: IBaseIconButton[];
  menu?: IBaseIconButton;
  onTitleClick?: () => void;
  title?: string;
  userMenu?: IBaseButton[];
  username?: string;
}
