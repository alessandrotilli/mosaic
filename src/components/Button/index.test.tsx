import React from 'react';
import { mount } from 'enzyme';
import { ButtonType } from '../../types/Button';
import Button from '.';

const defaultProps = {
  dataCy: 'myButton',
  label: 'Click Me',
  onClick: jest.fn(),
};

const componentWrapper = (props = {}) => (
  <Button
    {...defaultProps}
    {...props}
  />
)

describe('Button test suite:', () => {
  it('default', () => {
    const onClickHandler = jest.fn();
    const component = componentWrapper({
      onClick: onClickHandler,
    });
    const wrapper = mount(component);
    const button = wrapper.find('button');
    expect(button.prop('data-cy')).toEqual(defaultProps.dataCy);
    button.simulate('click');
    expect(onClickHandler).toHaveBeenCalledTimes(1);
    const buttonLabel = button.find('span.MuiButton-label');
    expect(buttonLabel.text()).toEqual(defaultProps.label);
  })

  it('failing test', () => {
    const onClickHandler = jest.fn();
    const component = componentWrapper({
      onClick: onClickHandler,
    });
    const wrapper = mount(component);
    const button = wrapper.find('button');
    const buttonLabel = button.find('span.MuiButton-label');
    expect(buttonLabel.text()).toEqual('Failing test');
  })
})