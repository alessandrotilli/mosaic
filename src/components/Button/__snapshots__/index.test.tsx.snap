// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suite: dataCy 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="custom"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Button test suite: default 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
</button>
`;

exports[`Button test suite: disabled 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
</button>
`;

exports[`Button test suite: elevated 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
</button>
`;

exports[`Button test suite: icon - custom 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-cy="button-icon"
      data-testid="StyleIcon"
      focusable="false"
      style={
        {
          "height": 24,
          "width": 24,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="m2.53 19.65 1.34.56v-9.03l-2.43 5.86c-.41 1.02.08 2.19 1.09 2.61zm19.5-3.7L17.07 3.98c-.31-.75-1.04-1.21-1.81-1.23-.26 0-.53.04-.79.15L7.1 5.95c-.75.31-1.21 1.03-1.23 1.8-.01.27.04.54.15.8l4.96 11.97c.31.76 1.05 1.22 1.83 1.23.26 0 .52-.05.77-.15l7.36-3.05c1.02-.42 1.51-1.59 1.09-2.6zM7.88 8.75c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-2 11c0 1.1.9 2 2 2h1.45l-3.45-8.34v6.34z"
      />
    </svg>
  </span>
  Button
</button>
`;

exports[`Button test suite: icon - left 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-cy="button-icon"
      data-testid="AccountCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
      />
    </svg>
  </span>
  Button
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Button test suite: icon - right 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
  <span
    className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-cy="button-icon"
      data-testid="AccountCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
      />
    </svg>
  </span>
</button>
`;

exports[`Button test suite: icon - rotate 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-rotate-2 css-i4bv87-MuiSvgIcon-root"
      data-cy="button-icon"
      data-testid="AccountCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
      />
    </svg>
  </span>
  Button
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Button test suite: icon 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-cy="button-icon"
      data-testid="AccountCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
      />
    </svg>
  </span>
  Button
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Button test suite: localized 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
  data-cy="locale.button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
</button>
`;

exports[`Button test suite: outlined 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1p5nen7-MuiButtonBase-root-MuiButton-root"
  data-cy="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
</button>
`;
